examples/LabeledMerge.tsx

```diff
@@ -0,0 +1,142 @@
+import * as React from 'react';
+import { useState, useEffect } from 'react';
+import { Range, getTrackBackground } from '../src/index';
+import { IRangeRef } from '../src/types';
+import { useThumbOverlap } from '../src/utils';
+
+const STEP = 0.1;
+const MIN = 0;
+const MAX = 100;
+const COLORS = ['#0C2960', '#276EF1', '#9CBCF8', '#ccc', 'red', 'green'];
+const THUMB_SIZE = 42;
+
+const ThumbLabel = ({
+  rangeRef,
+  values,
+  index
+}: {
+  rangeRef: IRangeRef | undefined;
+  values: number[];
+  index: number;
+}) => {
+  const [labelValue, style] = useThumbOverlap(rangeRef, values, index);
+  return (
+    <div
+      style={{
+        display: 'block',
+        position: 'absolute',
+        top: '-28px',
+        color: '#fff',
+        fontWeight: 'bold',
+        fontSize: '14px',
+        fontFamily: 'Arial,Helvetica Neue,Helvetica,sans-serif',
+        padding: '4px',
+        borderRadius: '4px',
+        backgroundColor: '#548BF4',
+        whiteSpace: 'nowrap',
+        ...(style as {})
+      }}
+    >
+      {labelValue}
+    </div>
+  );
+};
+class Labeled extends React.Component {
+  state: {
+    values: number[];
+    overlaps: any[];
+  } = {
+    values: [25, 50, 70, 80, 90],
+    overlaps: []
+  };
+  RANGE: any;
+  onChange = (values: number[]) => {
+    this.setState({
+      values
+    });
+  };
+  render() {
+    return (
+      <div
+        style={{
+          display: 'flex',
+          justifyContent: 'center',
+          flexWrap: 'wrap'
+        }}
+      >
+        <Range
+          allowOverlap
+          values={this.state.values}
+          ref={r => (this.RANGE = r)}
```

Comment:
`this.rangeRef` might be better naming? Also, `React.createRef()` seems to be the preferred way over callbacks since React `16.3`.